### PR TITLE
APP-3811 - Provisioning wrappers

### PR DESCRIPTION
**Pull request by clintpurser**
This is waiting on https://github.com/viamrobotics/api/pull/438

Add client wrappers and tests for the new provisioning API.

In its current state, this PR also adds generated code for provisioning that I manually generated from the above linked PR/branch. This won't actually get merged, but it is included here so I could write the wrappers and tests - and other work in the mobile app while I wait for that work to get finished.

### CODE REVIEW

**Comment by clintpurser** [comment]
@njooma i marked this as draft because we'll want to wait for the API change to come through first. But could you take a look at the client and the tests in the meantime? Thanks!

File: lib/src/app/provisioning.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  /// Get the status of the SmartMachine
+  Future<GetSmartMachineStatusResponse> getSmartMachineStatus() async {
+    final request = GetSmartMachineStatusRequest();
+    final response = await _client.getSmartMachineStatus(request);
+    return response;
+  }
+
+  /// Set the network credentials of the Smart machine, so it can connect to the internet.
+  ///
+  /// [type] is either 'wired' or 'wifi'
```
Comment (njooma) [inline lib/src/app/provisioning.dart:22]: Maybe make this an `enum` to make it strongly typed?

**Review by njooma — COMMENTED** [review]
Great start. Do you not have to also add this to the list of library exports though?

File: lib/src/app/provisioning.dart (hunk 1 above)
Comment (clintpurser) [reply lib/src/app/provisioning.dart:22]: good point, i thought about that but skipped for now. Will circle back and add in. Thanks!

**Comment by clintpurser** [comment]
> Great start. Do you not have to also add this to the list of library exports though?

ah yup, totally forgot that. Great catch, thanks!

File: lib/src/app/provisioning.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    final request = GetSmartMachineStatusRequest();
+    final response = await _client.getSmartMachineStatus(request);
+    return response;
+  }
+
+  /// Set the network credentials of the Smart machine, so it can connect to the internet.
+  ///
+  /// [ssid] is the networks ssid
+  /// [psk] is the networks passkey
+  Future<void> setNetworkCredentials({NetworkType type = NetworkType.wifi, required String ssid, required String psk}) async {
```
Comment (njooma) [inline lib/src/app/provisioning.dart:26]: `ssid` and `psk` shouldn't be required for `wired` connections. Not a locker, but please add a TODO

File: lib/src/app/provisioning.dart [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    final request = SetNetworkCredentialsRequest(type: type.name, ssid: ssid, psk: psk);
+    await _client.setNetworkCredentials(request);
+  }
+
+  /// Set the Viam credentials of the Smart machine, so it can connect to the Viam.
```
Comment (njooma) [inline lib/src/app/provisioning.dart:31]: > connect to the Viam

😂

**Review by njooma — APPROVED** [review]

**Pull request merged**